FreeAPS/Sources/Services/ContactTrick/ContactTrickManager.swift

```diff
@@ -0,0 +1,198 @@
+
+            self.renderContacts()
+        }
+    }
+
+    private func renderContacts() {
+        contacts.forEach { renderContact($0) }
+        workItem = DispatchWorkItem(block: {
+            print("in updateContact, no updates received for more than 5 minutes")
+            self.renderContacts()
+        })
+        DispatchQueue.main.asyncAfter(deadline: .now() + 5 * 60 + 15, execute: workItem!)
+    }
+
+    private func renderContact(_ entry: ContactTrickEntry) {
+        guard let contactId = entry.contactId, entry.enabled else {
+            return
+        }
+
+        let keysToFetch = [CNContactImageDataKey] as [CNKeyDescriptor]
+
+        let contact: CNContact
+        do {
+            contact = try contactStore.unifiedContact(withIdentifier: contactId, keysToFetch: keysToFetch)
+        } catch {
+            print("in updateContact, an error has been thrown while fetching the selected contact")
+            return
+        }
+
+        guard let mutableContact = contact.mutableCopy() as? CNMutableContact else {
+            return
+        }
+
+        mutableContact.imageData = ContactPicture.getImage(
+            contact: entry,
+            state: state
+        ).pngData()
+
+        let saveRequest = CNSaveRequest()
+        saveRequest.update(mutableContact)
+        do {
+            try contactStore.execute(saveRequest)
+        } catch {
+            print("in updateContact, failed to update the contact")
+        }
+    }
+
+    private func glucoseText(_ glucose: [Readings]) -> (glucose: String, trend: String, delta: String) {
+        let glucoseValue = glucose.first?.glucose ?? 0
+
+        guard !glucose.isEmpty else { return ("--", "--", "--") }
+
+        let delta = glucose.count >= 2 ? glucoseValue - glucose[1].glucose : nil
+
+        let units = settingsManager.settings.units
+        let glucoseText = glucoseFormatter
+            .string(from: Double(
+                units == .mmolL ? Decimal(glucoseValue).asMmolL : Decimal(glucoseValue)
+            ) as NSNumber)!
+
+        let directionText = glucose.first?.direction ?? "↔︎"
+        let deltaText = delta
+            .map {
+                self.deltaFormatter
+                    .string(from: Double(
+                        units == .mmolL ? Decimal($0).asMmolL : Decimal($0)
+                    ) as NSNumber)!
+            } ?? "--"
+
+        return (glucoseText, directionText, deltaText)
+    }
+
+    private func eventualBGString() -> String? {
+        guard let eventualBG = suggestion?.eventualBG else {
+            return nil
+        }
+        let units = settingsManager.settings.units
+        return eventualFormatter.string(
+            from: (units == .mmolL ? eventualBG.asMmolL : Decimal(eventualBG)) as NSNumber
+        )!
+    }
+
+    private var glucoseFormatter: NumberFormatter {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        formatter.locale = Locale(identifier: "en_US")
```

Comment:
I added the locale explicitly here in order to make it print decimal separators as points (`.`) not commas (`,`), because It looked weird-ish on the complication, but it doesn't seem right to override the locale like this 🤔